### DISCUSSION TITLE
Newbie questions: updating and best uses of homebrew

### DISCUSSION THREAD

**Discussion by mbierman** [Homebrew/discussions#1912] [Getting started] [answered]
Hi I've been using brew for a while but consider myself somewhat a newbie. My apologies in advance.

1. What is the best way to keep things updated? I have the following simple script which does the job, but I'm realizing may be unnecessary. 

```
brew update
brew outdated
brew upgrade
brew cleanup
brew upgrade --cask aerial
...
```
do I need to update individual casks? Or is that redundant? 

The other questions are more meta.
2. How secure are homebrew packages? For example, what prevents someone from modifying a trusted app like say, Firefox and creating a cask with a tainted version?
3. What happens if you install a cask but it is updated by the application itself (e.g. firefox) ? Does this create any conflicts? 
4. Why do people prefer to install something like Firefox via homebrew vs a simple download? Is it just so you can easily script setting up a new machine or are there other benefits?
5. If I install a GUI app via homebrew (e.g. firefox) and want to uninstall, can I just drag to the trash or do I need to do something in homebrew instead/as well?

**Comment by SMillerDev** [comment, marked as answer]
> What is the best way to keep things updated?

`brew upgrade` will update everything it can safety update.

> do I need to update individual casks? Or is that redundant?

Casks will be updated by `brew upgrade` unless they are auto updating.

> 2. How secure are homebrew packages? For example, what prevents someone from modifying a trusted app like say, Firefox and creating a cask with a tainted version?

Nothing prevents people from using homebrew to distribute tainted software, but for the official repositories everything is reviewed by humans to prevent that.

> 3. What happens if you install a cask but it is updated by the application itself (e.g. firefox) ? Does this create any conflicts?

It would, that's why those casks don't upgrade through brew by default. You can force it using `brew upgrade --greedy` though.

> 4. Why do people prefer to install something like Firefox via homebrew vs a simple download? Is it just so you can easily script setting up a new machine or are there other benefits?

I can't speak for everyone, but for me it's this + easily seeing when there are updated versions.

> 5. If I install a GUI app via homebrew (e.g. firefox) and want to uninstall, can I just drag to the trash or do I need to do something in homebrew instead/as well?

If you install it through homebrew, you should use homebrew to remove it again. Otherwise homebrew might reinstall it on the next package update.

**Reply by mbierman** [reply]
Many thanks for your time and patience @SMillerDev.
